userprofile/views.py

```diff
@@ -47,6 +56,48 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class LanguageView():
+    """"This language view class overrides set_language function from 
+    django.views.i18n."""
+        
+    def set_language(request):
+        LANGUAGE_QUERY_PARAMETER = 'language'
```

Comment:
I think it would be a good idea to rename this so that there isn't word `QUERY` in it. It's likely that when removing the cookie, the language is set as a query parameter (e.g. `url + '&hl=' + lang`).